File: app/src/main/java/com/skydoves/waterdays/ApplicationModule.java
package com.skydoves.waterdays;

import android.app.Application;
import android.support.annotation.NonNull;

import com.skydoves.waterdays.persistence.preference.PreferenceManager;
import com.skydoves.waterdays.persistence.sqlite.SqliteManager;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Developed by skydoves on 2017-08-19.
 * Copyright (c) 2017 skydoves rights reserved.
 */

@Module
public final class ApplicationModule {
    private final Application application;

    public ApplicationModule(final @NonNull Application application) {
        this.application = application;
    }

    @Provides
    @Singleton
    @NonNull PreferenceManager providePreferenceManager() {
        return new PreferenceManager(this.application);
    }

    @Provides
    @Singleton
    @NonNull SqliteManager provideSqliteManager() {
        return new SqliteManager(this.application, SqliteManager.DATABASE_NAME, null, SqliteManager.DATABASE_VERSION);
    }
}

File: app/src/main/java/com/skydoves/waterdays/ApplicationModule.kt
package com.skydoves.waterdays

import android.app.Application
import com.skydoves.waterdays.persistence.preference.PreferenceManager
import com.skydoves.waterdays.persistence.sqlite.SqliteManager
import com.skydoves.waterdays.utils.AlarmUtils
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * Developed by skydoves on 2017-08-19.
 * Copyright (c) 2017 skydoves rights reserved.
 */

@Module
class ApplicationModule(private val application: Application) {

  @Provides
  @Singleton
  internal fun providePreferenceManager(): PreferenceManager {
    return PreferenceManager(this.application)
  }

  @Provides
  @Singleton
  internal fun provideSqliteManager(): SqliteManager {
    return SqliteManager(this.application,
        SqliteManager.DATABASE_NAME,
        null,
        SqliteManager.DATABASE_VERSION)
  }

  @Provides
  @Singleton
  internal fun provideAlarmUtils(): AlarmUtils {
    return AlarmUtils(this.application)
  }
}
